apps/charterafrica/src/lib/data/common/processPageDatasets.js

```diff
@@ -1,21 +1,38 @@
 import fetchDatasets from "@/charterafrica/lib/openAfrica";
+import datasetsQuery from "@/charterafrica/utils/datasets/queryString";
 
-async function processPageDatasets(page, api) {
+const getDatasetsQuery = (context) => {
+  const { query = {}, locale } = context;
+  const { page = 1, q, tags = "", countries = "", sort = "" } = query;
+
+  return { countries, locale, page, q, tags, sort };
+};
+
+export default async function processPageDatasets(page, api, context) {
   const { blocks } = page;
+  const { organizationId } = await api.findGlobal("openAfrica");
+
+  const data = await fetchDatasets(organizationId);
+  const { count, datasets, countries, tags, totalPages } = data;
```

Comment:
1. Move this data fetching to the inside of `if(datasetsIndex > -1)`. Why fetch data if there isn't `datasets` block?
2. Also, only fetch data if `organizationId` is set.